src/dlgRoomExits.cpp

```diff
@@ -758,6 +897,54 @@ void dlgRoomExits::save()
+        int exitAreaID = pExitToRoom->getArea();
+        bool outOfAreaExit = (exitAreaID && exitAreaID != mAreaID);
+        QString exitAreaName;
+        if (outOfAreaExit) {
+            exitAreaName = mpHost->mpMap->mpRoomDB->getAreaNamesMap().value(exitAreaID);
+        }
+
+        // This is the toolTip text for the room ID number column (and the
+        // status icon)
+        const QString roomIdToolTipText{generateToolTip(pExitToRoom->name, exitAreaName, outOfAreaExit, pExitToRoom->getWeight())};
+        pSpecialExit->setIcon(ExitsTreeWidget::colIndex_exitStatus, showIconOnExitStatus ? (outOfAreaExit ? mIcon_otherAreaExit : mIcon_inAreaExit) : QIcon());
+        pSpecialExit->setToolTip(ExitsTreeWidget::colIndex_exitRoomId, roomIdToolTipText);
+        pSpecialExit->setToolTip(ExitsTreeWidget::colIndex_exitStatus, roomIdToolTipText);
+
+    } else if (pSpecialExit->text(ExitsTreeWidget::colIndex_exitRoomId).toInt() > 0) {
+        // A number but not valid
+        pSpecialExit->setIcon(ExitsTreeWidget::colIndex_exitStatus, showIconOnExitStatus ? mIcon_invalidExit : QIcon());
+        pSpecialExit->setToolTip(ExitsTreeWidget::colIndex_exitRoomId, singleParagraph.arg(tr("Entered number is invalid, set the number of the room that this special exit leads to; "
+                                                                                              "if left like this, this exit will be deleted when <tt>save</tt> is clicked.")));
+    } else if (pSpecialExit->text(ExitsTreeWidget::colIndex_exitRoomId).isEmpty() || pSpecialExit->text(ExitsTreeWidget::colIndex_exitRoomId) == mSpecialExitRoomIdPlaceholder) {
+        // Nothing
+        pSpecialExit->setIcon(ExitsTreeWidget::colIndex_exitStatus, QIcon());
+        pSpecialExit->setToolTip(ExitsTreeWidget::colIndex_exitRoomId, singleParagraph.arg(tr("A positive room ID of the room that this special exit leads to is expected here; "
+                                                                                              "if left like this, this exit will be deleted when <tt>save</tt> is clicked.")));
+    } else {
+        // Something else that isn't a positive number
+        pSpecialExit->setIcon(ExitsTreeWidget::colIndex_exitStatus, showIconOnExitStatus ? mIcon_invalidExit : QIcon());
+        pSpecialExit->setToolTip(ExitsTreeWidget::colIndex_exitRoomId, singleParagraph.arg(tr("A positive room ID of the room that this special exit leads to is expected here; "
+                                                                                              "if left like this, this exit will be deleted when <tt>save</tt> is clicked.")));
+    }
+
+    if (pSpecialExit->text(ExitsTreeWidget::colIndex_command) == mSpecialExitCommandPlaceholder) {
+        pSpecialExit->setToolTip(ExitsTreeWidget::colIndex_command, singleParagraph.arg(tr("No command or Lua script entered, if left like this, this exit will be deleted when <tt>save</tt> is clicked.")));
```

Comment:
When there are two separate ideas, don't put a comma instead of full stop.
```suggestion
        pSpecialExit->setToolTip(ExitsTreeWidget::colIndex_command, singleParagraph.arg(tr("No command or Lua script entered. If left like this, this exit will be deleted when <tt>save</tt> is clicked.")));
```
The ideas "No command or Lua script entered, if left like this" did not have any connection intended, did they?

Reply:
Actually the ideas *are* connected - the absence of a command or script entry will, if not corrected, cause deletion.

Reply:
Yes but read that part again. It says: If you leave it like this, nothing is entered. They are not connected that way, no?